DNN Platform/Library/Services/Localization/LocalizationProvider.cs

```diff
@@ -97,7 +97,19 @@ public string GetString(string key, string resourceFileRoot, string language, Po
                 Logger.WarnFormat("Missing localization key. key:{0} resFileRoot:{1} threadCulture:{2} userlan:{3}", key, resourceFileRoot, Thread.CurrentThread.CurrentUICulture, language);
             }
 
-            return string.IsNullOrEmpty(resourceValue) ? string.Empty : resourceValue;
+            return string.IsNullOrEmpty(resourceValue) ? string.Empty : RemoveHttpUrlsIfSiteisSSLEnabled(resourceValue);
+        }
+
+        private string RemoveHttpUrlsIfSiteisSSLEnabled(string resourceValue)
+        {
+            var portalSettings = PortalController.Instance.GetCurrentPortalSettings();
+
+            if (portalSettings.SSLEnabled)
+            {
+                resourceValue = resourceValue.Replace(@"http:", @"https:");
```

Comment:
I think resource entries should not be containing the external URLs. Either pointing to parent or child portals, which is taken care by GetCurrentPortalSettings

Reply:
I agree. But your check is still missing one check for the other setting
``if (portalSettings.SSLEnabled || portalSettings.SSLEnforced)``

Reply:
Ok thnx. Put that as well.